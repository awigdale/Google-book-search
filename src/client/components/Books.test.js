import React from 'react';
import { shallow, mount } from 'enzyme';
import Books from './Books';

describe('Books component', () => {
  it('has state', () => {
    const wrapper = mount(<Books />);
    expect(wrapper.state('searchInput')).toEqual('');
    expect(wrapper.state('allBooks')).toHaveLength(0);
  });
  it('should have a functional submit button', () => {
    const wrapper = mount(<Books />);
    const handleSubmit = spyOn(wrapper.instance(), 'handleSubmit');
    wrapper.setState({ searchInput: 'harry potter' });
    wrapper.find('button').simulate('click');
    expect(handleSubmit).toHaveBeenCalled();
  });
  // it('renders a list of books', () => {
  //   const wrapper = mount(<Books />);
  //   wrapper.setState({
  //     allBooks: [
  //       { volumeInfo: { title: 'harry potter' } },
  //       { volumeInfo: { title: 'harry potter2' } },
  //       { volumeInfo: { title: 'harry potter3' } },
  //     ],
  //   });
  //   expect(wrapper.hasClass('book_container'));
  // });
});
