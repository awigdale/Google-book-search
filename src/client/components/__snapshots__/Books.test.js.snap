// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book has state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Books />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <form>
          <input
            onChange={[Function]}
            placeholder="Search by author, title, keywords"
            type="text"
            value=""
          />
          <button
            id="submit"
            onClick={[Function]}
          >
            search
          </button>
        </form>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Search by author, title, keywords"
              type="text"
              value=""
            />,
            <button
              id="submit"
              onClick={[Function]}
            >
              search
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Search by author, title, keywords",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "search",
              "id": "submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <form>
            <input
              onChange={[Function]}
              placeholder="Search by author, title, keywords"
              type="text"
              value=""
            />
            <button
              id="submit"
              onClick={[Function]}
            >
              search
            </button>
          </form>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Search by author, title, keywords"
                type="text"
                value=""
              />,
              <button
                id="submit"
                onClick={[Function]}
              >
                search
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Search by author, title, keywords",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "search",
                "id": "submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "search",
              "type": "button",
            },
          ],
          "type": "form",
        },
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Books component renders without state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Books />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <form>
          <input
            onChange={[Function]}
            placeholder="Search by author, title, keywords"
            type="text"
            value=""
          />
          <button
            id="submit"
            onClick={[Function]}
          >
            search
          </button>
        </form>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Search by author, title, keywords"
              type="text"
              value=""
            />,
            <button
              id="submit"
              onClick={[Function]}
            >
              search
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Search by author, title, keywords",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "search",
              "id": "submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <form>
            <input
              onChange={[Function]}
              placeholder="Search by author, title, keywords"
              type="text"
              value=""
            />
            <button
              id="submit"
              onClick={[Function]}
            >
              search
            </button>
          </form>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Search by author, title, keywords"
                type="text"
                value=""
              />,
              <button
                id="submit"
                onClick={[Function]}
              >
                search
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Search by author, title, keywords",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "search",
                "id": "submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "search",
              "type": "button",
            },
          ],
          "type": "form",
        },
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
